uprotocol/transport/builder/umessagebuilder.py

```diff
@@ -0,0 +1,318 @@
+        @return Returns the UMessageBuilder with the configured priority
+        and sink.
+        """
+        if source is None:
+            raise ValueError(SOURCE_ERROR)
+        if sink is None:
+            raise ValueError(SINK_ERROR)
+        return UMessageBuilder(
+            source,
+            Factories.UPROTOCOL.create(),
+            UMessageType.UMESSAGE_TYPE_NOTIFICATION,
+        ).with_sink(sink)
+
+    @staticmethod
+    def request(source: UUri, sink: UUri, ttl: int):
+        """Construct a UMessageBuilder for a request message.
+
+        @param source   Source address of the message.
+        @param sink The destination URI.
+        @param ttl The time to live in milliseconds.
+        @return Returns the UMessageBuilder with the configured priority,
+        sink and ttl.
+        """
+        if source is None:
+            raise ValueError(SOURCE_ERROR)
+        if sink is None:
+            raise ValueError(SINK_ERROR)
+        if ttl is None:
+            raise ValueError(TTL_ERROR)
+        return (
+            UMessageBuilder(
+                source,
+                Factories.UPROTOCOL.create(),
+                UMessageType.UMESSAGE_TYPE_REQUEST,
+            )
+            .with_ttl(ttl)
+            .with_sink(sink)
+        )
+
+    @staticmethod
+    def response(source: UUri, sink: UUri, reqid: UUID):  # noqa: N805
+        """Construct a UMessageBuilder for a response message.
+
+        @param source   Source address of the message.
+        @param sink The destination URI.
+        @param reqid The original request UUID used to correlate the
+        response to the request.
+        @return Returns the UMessageBuilder with the configured source,
+        sink, priority, and reqid.
+        """
+        if source is None:
+            raise ValueError(SOURCE_ERROR)
+        if sink is None:
+            raise ValueError(SINK_ERROR)
+        if reqid is None:
+            raise ValueError(REQID_ERROR)
+
+        return (
+            UMessageBuilder(
+                source,
+                Factories.UPROTOCOL.create(),
+                UMessageType.UMESSAGE_TYPE_RESPONSE,
+            )
+            .with_sink(sink)
+            .with_reqid(reqid)
+        )
+
+    @staticmethod
+    def response_for_request(request: UAttributes):  # noqa: N805
+        """Construct a UMessageBuilder for a response message using an
+        existing request.
+        @param request The original request UAttributes used to correlate
+        the response to the request.
+        @return Returns the UMessageBuilder with the configured source, sink,
+        priority, and reqid.
+        """
+        if request is None:
+            raise ValueError(REQUEST_ERROR)
+        return (
+            UMessageBuilder(
+                request.sink,
+                Factories.UPROTOCOL.create(),
+                UMessageType.UMESSAGE_TYPE_RESPONSE,
+            )
+            .with_priority(UPriority.UPRIORITY_CS4)
```

Comment:
```suggestion
            .with_priority(request.priority)
```

Reply:
Fixed it and added a test case to verify this scenario.